adapter/dubbogo/consumer_filter.go

```diff
@@ -0,0 +1,50 @@
+package dubbogo
+
+import (
+	"github.com/apache/dubbo-go/protocol"
+)
+import (
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+)
+
+type consumerFilter struct{}
+
+func (d *consumerFilter) Invoke(invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	methodResourceName := getResourceName(invoker, invocation, getConsumerPrefix())
+	interfaceResourceName := ""
+	if getInterfaceGroupAndVersionEnabled() {
+		interfaceResourceName = getColonSeparatedKey(invoker.GetUrl())
+	} else {
+		interfaceResourceName = invoker.GetUrl().Service()
+	}
+	if !isAsync(invocation) {
+		_, b := sentinel.Entry(interfaceResourceName,
+			sentinel.WithResourceType(base.ResTypeRPC),
+			sentinel.WithTrafficType(base.Outbound))
+		if b != nil { // blocked
+			result := &protocol.RPCResult{}
+			result.SetResult(nil)
+			result.SetError(b)
+			return result
+		}
+		_, b = sentinel.Entry(methodResourceName,
+			sentinel.WithResourceType(base.ResTypeRPC),
+			sentinel.WithTrafficType(base.Outbound),
+			sentinel.WithArgs(invocation.Attachments()))
+		if b != nil { // blocked
+			result := &protocol.RPCResult{}
+			result.SetResult(nil)
+			result.SetError(b)
+			return result
+		}
+	} else {
+		// todo : Need to implement asynchronous current limiting
+		//  unlimited flow for the time being
+	}
+	return invoker.Invoke(invocation)
+}
+
+func (d *consumerFilter) OnResponse(result protocol.Result, _ protocol.Invoker, _ protocol.Invocation) protocol.Result {
+	return result
```

Comment:
Actually, when the invocation completes (whether successful or with error), we need to call `e.Exit()` to mark it as completed, or the response time, complete count and error count will be lost. We may need to carry the `*SentinelEntry` in the RPC context (if possible) and handle it in `OnResponse` callback.

FYI: [onResponse logic in Java version (1.7.1)](https://github.com/alibaba/Sentinel/blob/1.7.1/sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/BaseSentinelDubboFilter.java#L45)

Reply:
please check it